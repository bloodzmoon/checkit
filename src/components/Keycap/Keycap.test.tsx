import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { Keycap } from '@components'
import { FiCheck } from 'react-icons/fi'

describe('<Keycap>', () => {
  it('should render <span>', () => {
    const wrapper = shallow(<Keycap />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render `string child` correctly', () => {
    const expectedText = 'Shift'
    const wrapper = shallow(<Keycap>{expectedText}</Keycap>)
    const child = wrapper.children()
    expect(child.text()).toBe(expectedText)
  })

  it('should render `icon child` correctly', () => {
    const wrapper = shallow(
      <Keycap>
        <FiCheck />
      </Keycap>
    )
    const child = wrapper.children()
    expect(child.type()).toBe(FiCheck)
  })

  it('should render class from props correctly', () => {
    const wrapper = shallow(<Keycap size="1.5" halfHeight />)
    expect(wrapper.hasClass('size1-5')).toBeTruthy()
    expect(wrapper.hasClass('half')).toBeTruthy()
    expect(wrapper.hasClass('gap')).toBeTruthy()
  })

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Keycap size="1.15" halfHeight>
          Test
        </Keycap>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
