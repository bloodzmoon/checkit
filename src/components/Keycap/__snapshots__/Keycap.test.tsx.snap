// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keycap> should match snapshot 1`] = `
<span
  className="base size1-15 half"
>
  Test
</span>
`;
